mm2src/coins/lightning/ln_events.rs

```diff
@@ -126,44 +164,64 @@ fn sign_funding_transaction(
 
 impl LightningEventHandler {
     pub fn new(
-        filter: Arc<PlatformFields>,
+        platform: Arc<Platform>,
         channel_manager: Arc<ChannelManager>,
         keys_manager: Arc<KeysManager>,
-        inbound_payments: PaymentsMapShared,
-        outbound_payments: PaymentsMapShared,
+        persister: Arc<LightningPersister>,
     ) -> Self {
         LightningEventHandler {
-            filter,
+            platform,
             channel_manager,
             keys_manager,
-            inbound_payments,
-            outbound_payments,
+            persister,
         }
     }
 
-    fn handle_funding_generation_ready(&self, temporary_channel_id: [u8; 32], output_script: &Script) {
+    fn handle_funding_generation_ready(
+        &self,
+        temporary_channel_id: [u8; 32],
+        channel_value_satoshis: u64,
+        output_script: &Script,
+        user_channel_id: u64,
+    ) {
         log::info!(
-            "Handling FundingGenerationReady event for temporary_channel_id: {}",
-            hex::encode(temporary_channel_id)
+            "Handling FundingGenerationReady event for internal channel id: {}",
+            user_channel_id
         );
-        let funding_tx = match sign_funding_transaction(temporary_channel_id, output_script, self.filter.clone()) {
+        let funding_tx = match sign_funding_transaction(user_channel_id, output_script, self.platform.clone()) {
             Ok(tx) => tx,
             Err(e) => {
                 log::error!(
-                    "Error generating funding transaction for temporary channel id {:?}: {}",
-                    temporary_channel_id,
+                    "Error generating funding transaction for internal channel id {}: {}",
+                    user_channel_id,
                     e.to_string()
                 );
                 return;
             },
         };
+        let funding_txid = funding_tx.txid();
         // Give the funding transaction back to LDK for opening the channel.
         if let Err(e) = self
             .channel_manager
             .funding_transaction_generated(&temporary_channel_id, funding_tx)
         {
             log::error!("{:?}", e);
+            return;
         }
+        let platform = self.platform.clone();
+        let persister = self.persister.clone();
+        spawn(async move {
+            let current_block = platform.coin.current_block().compat().await.unwrap_or_default();
```

Comment:
I'm not sure that default here is justified, is it's the case, you will add a tx to sql with current_block equal to zero

Reply:
Well, current_block's only purpose is to use it later here https://github.com/KomodoPlatform/atomicDEX-API/blob/6b2e98fda5775cd857e69a2438f80d6bc0a552f7/mm2src/coins/lightning/ln_events.rs#L368 to get the closing tx, using 0 for the `from_block` param shouldn't be a problem if `platform.coin.current_block()` somehow failed.

I can also see that `from_block` is not really used in `find_output_spend` if electrum is enabled, but using 0 for the `from_block` param can be a problem if native mode is enabled since all transactions will be listed from the first block but that will only be to happen only as a contingency plan if `platform.coin.current_block()` somehow failed.

Reply:
`funding_generated_in_block` looks a bit confusing to me. Does it really mean that transaction was generated during this block, or it's assumed to be *confirmed* in this block?

Reply:
Using the default 0 value seems problematic for the native mode in the future. We get the current block number during each coin activation, what do you think about using this value instead of 0? It will be an additional `activated_at_block` field in the `LightningCoin`.

Reply:
Also, this value can be updated, when transaction is confirmed.

Reply:
>  We get the current block number during each coin activation, what do you think about using this value instead of 0? It will be an additional activated_at_block field in the LightningCoin.

I guess `best_block_height` can be added to the `Platform` struct https://github.com/KomodoPlatform/atomicDEX-API/blob/48256014a5cd8e9a2ee2b1cce06a120091231a8d/mm2src/coins/lightning/ln_platform.rs#L209-L222
and updated before https://github.com/KomodoPlatform/atomicDEX-API/blob/48256014a5cd8e9a2ee2b1cce06a120091231a8d/mm2src/coins/lightning/ln_platform.rs#L148